zio-http/src/main/scala/zio/http/codec/internal/BodyCodec.scala

```diff
@@ -66,6 +78,11 @@ private[internal] sealed trait BodyCodec[A] { self =>
    */
   def mediaType: Option[MediaType]
 
+  /**
+   * Returns the media type or application/json if not specified
+   */
+  def mediaTypeOrJson: MediaType = mediaType.getOrElse(MediaType.application.`json`)
```

Comment:
Having this on `BodyCodec` is a bit weird to me. Maybe it would be nicer to just have the optional `mediaType` here, and have the fallback value specified in `EncoderDecoder` (seems like that's the only place this is used anyway)